state/accountsRepository.go

```diff
@@ -49,16 +49,15 @@ func (repository *accountsRepository) GetAccountWithBlockInfo(address []byte, op
 		return nil, nil, err
 	}
 
-	convertedOptions, err := repository.convertAccountQueryOptions(options)
-	if err != nil {
-		return nil, nil, err
-	}
-
+	convertedOptions := repository.convertAccountQueryOptions(options)
 	return accountsAdapter.GetAccountWithBlockInfo(address, convertedOptions)
 }
 
-func (repository *accountsRepository) convertAccountQueryOptions(options api.AccountQueryOptions) (common.RootHashHolder, error) {
-	return holders.NewRootHashHolder(options.BlockRootHash), nil
+func (repository *accountsRepository) convertAccountQueryOptions(options api.AccountQueryOptions) common.RootHashHolder {
+	if options.HintEpoch.HasValue {
```

Comment:
instead of this if, you could have send the optionalUint32 parameter in the NewRootHashHolder constructor

Reply:
Fixed.